plugins/MauticCrmBundle/Integration/Pipedrive/Import/CompanyImport.php

```diff
@@ -85,6 +97,8 @@ public function update(array $data = [])
         $integrationEntity = $this->getCompanyIntegrationEntity(['integrationEntityId' => $data['id']]);
 
         if (!$integrationEntity) {
+            sleep(3);
```

Comment:
If your create contact on Pipedrive, the service send two webhooks:
First create contact
Second update contact

During the process, sometimes integegration_entity  was not saved before plugin process  second webhook and create multiple contacts

Reply:
It turns out the 2 webhooks at the same time problem can be fixed in the configuration of the Pipedrive webhooks. I updated the docs: https://github.com/mautic/documentation/pull/290/files

What do you think? Can we remove the sleep funciton if the webhooks will be set only to update?

Reply:
I think It doesn't  work with added and update event action with person/company, does it?
Or do you mean use just update event action and added event action is not necessary then?

Reply:
Yep, the update action is all we need. It's triggered on create too. Adding the create action to it causes the duplicates.

Reply:
@npracht needs you.
What do you think about docs improvement and use just update action for webhook. It's same results, just old clients needs change webhook settings.

Reply:
If it works same or better, i think use the new method. @escopecz you'll need to highlight a special message in the Release note i guess.

Reply:
This is what I got from the Pipedrive support:

> We would suggest for you to do it this way:
> 
> 1. on person.added just insert data to their DB
> 2. on person.updated try to update existing person on your side (using meta.id as a unique constraint)
> 2.1 if there is nothing on your side yet, respond to Webhook request with 500 Server Error and Pipedrive will try to deliver this event again in 3 sec (https://pipedrive.readme.io/docs/guide-for-webhooks#section-retry-logic) 
> 
> This way you will not create duplicates on your side. Let me know if you have further questions please.

But this is tricky to do with ORM...

Reply:
The same shit happend If we 're trying create contact and with comapny in Pipedrive.
Both hooks (company update, contact update) are created at same time and during the create contact We cannot addCompanyToLead because because company entity is not exist at that time.

Maybe we could return 500error and wait 3 sec for retry request (but It's ugly)
Do you have another idea?

Reply:
Or, how about to ignore all webhooks type of create?

Reply:
I removed create hooks, and first test looks good. Works. Thanks